packages/stream-metadata/src/routes/spaceMetadata.ts

```diff
@@ -1,36 +1,128 @@
+import { getStream } from '../riverStreamRpcClient'
 import { isValidEthereumAddress } from '../validators'
+import { spaceDapp } from '../contract-utils'
 
-export function fetchSpaceMetadata(
-	request: FastifyRequest,
-	reply: FastifyReply,
-	serverUrl: string,
-) {
+const paramsSchema = z.object({
+	spaceAddress: z.string().min(1, 'spaceAddress parameter is required'),
+})
+
+export interface SpaceMetadataResponse {
+	name: string
+	description: string
+	image: string | undefined
+}
+
+export async function fetchSpaceMetadata(request: FastifyRequest, reply: FastifyReply) {
 	const logger = request.log.child({ name: fetchSpaceMetadata.name })
-	const { spaceAddress } = request.params as { spaceAddress?: string }
-	logger.info({ spaceAddress }, 'GET /space/../metadata')
 
-	if (!spaceAddress) {
-		return reply
-			.code(400)
-			.send({ error: 'Bad Request', message: 'spaceAddress parameter is required' })
+	const parseResult = paramsSchema.safeParse(request.params)
+
+	if (!parseResult.success) {
+		const errorMessage = parseResult.error.errors[0]?.message || 'Invalid parameters'
+		logger.info(errorMessage)
+		return reply.code(400).send({ error: 'Bad Request', message: errorMessage })
 	}
 
+	const { spaceAddress } = parseResult.data
+
 	// Validate spaceAddress format using the helper function
 	if (!isValidEthereumAddress(spaceAddress)) {
+		logger.info({ spaceAddress }, 'Invalid spaceAddress format')
 		return reply
 			.code(400)
 			.send({ error: 'Bad Request', message: 'Invalid spaceAddress format' })
 	}
 
-	const dummyJson = {
-		name: '....',
-		description: '....',
-		members: 99999,
-		fees: '0.001 eth',
-		image: `${serverUrl}/space/${spaceAddress}/image`,
+	let spaceInfo: SpaceInfo | undefined
+	try {
+		spaceInfo = await spaceDapp.getSpaceInfo(spaceAddress)
+	} catch (error) {
+		logger.error({ spaceAddress, error }, 'Failed to fetch space contract info')
+		return reply
+			.code(404)
+			.send({ error: 'Not Found', message: 'Failed to fetch space contract info' })
+	}
+
+	if (!spaceInfo) {
+		logger.error({ spaceAddress }, 'Space contract not found')
+		return reply.code(404).send({ error: 'Not Found', message: 'Space contract not found' })
+	}
+
+	const spaceMetadata: SpaceMetadataResponse = {
+		name: spaceInfo.name,
+		description: getSpaceDecription(spaceInfo),
+		image: await getImageUrl(logger, spaceInfo.uri, spaceAddress),
+	}
+
+	return reply.header('Content-Type', 'application/json').send(spaceMetadata)
+}
+
+function getSpaceDecription({ shortDescription, longDescription }: SpaceInfo): string {
+	if (shortDescription && longDescription) {
+		return `${shortDescription}<br><br>${longDescription}`
+	}
+
+	if (shortDescription) {
+		return shortDescription
+	}
+
+	return longDescription || ''
+}
+
+async function getImageUrl(logger: FastifyBaseLogger, contractUri: string, spaceAddress: string) {
+	const hasSpaceImageExist = await hasSpaceImage(logger, spaceAddress)
+	if (!hasSpaceImageExist) {
+		return undefined
+	}
```

Comment:
we're allowing people to overwrite the contract uri right? so we can't return undefined here. I'm not sure why we need to check if it's set at all, seems like we should handle not found when you call the /image url. 
what exactly are you trying to achieve?

Reply:
good point. Will remove